tests/fsm/test_json_schema.py

```diff
@@ -1,65 +1,62 @@
 import json
 import re
-from typing import Literal, Union
 
 import pytest
-from outlines_core.fsm.json_schema import build_regex_from_schema, to_regex
-from pydantic import BaseModel, Field
-
-
-@pytest.mark.parametrize("whitespace_pattern", [None, r"[\n ]*", "abc"])
-def test_json_schema_custom_whitespace_pattern(whitespace_pattern):
-    """assert whitespace_pattern setting respected"""
-
-    class MockModel(BaseModel):
+from outlines_core.fsm.json_schema import (  # noqa: F401
+    BOOLEAN,
+    DATE,
+    DATE_TIME,
+    EMAIL,
+    INTEGER,
+    NULL,
+    NUMBER,
+    STRING,
+    STRING_INNER,
+    TIME,
+    URI,
+    UUID,
+    WHITESPACE,
+    build_regex_from_schema,
+)
+from pydantic import BaseModel
+
+
+def test_build_regex_from_json_schema():
+    class FooBar(BaseModel):
         foo: int
         bar: str
 
-    schema = json.dumps(MockModel.model_json_schema())
-
-    # assert any ws pattern can be used
-    if whitespace_pattern == "abc":
-        build_regex_from_schema(schema, whitespace_pattern)
-        return
-
-    pattern = build_regex_from_schema(schema, whitespace_pattern)
-
-    mock_result_mult_ws = (
-        """{     "foo"   :   4, \n\n\n   "bar": "baz    baz baz bar"\n\n}"""
-    )
-    mock_result_maybe_ws = """{"foo" : 4 ,"bar":"baz    baz baz bar"}"""
-
-    match_default_ws = re.fullmatch(pattern, mock_result_maybe_ws)
-    if whitespace_pattern is None:
-        assert match_default_ws
-    else:
-        assert re.fullmatch(pattern, mock_result_mult_ws)
-
-
-def test_one_of_doesnt_produce_illegal_lookaround():
-    """Reproduces failure in https://github.com/dottxt-ai/outlines/issues/823"""
-
-    class Cat(BaseModel):
-        pet_type: Literal["cat"]
-        meows: int
-
-    class Dog(BaseModel):
-        pet_type: Literal["dog"]
-        barks: float
-
-    class Model(BaseModel):
-        pet: Union[Cat, Dog] = Field(..., discriminator="pet_type")
-        n: int
-
-    json_schema = json.dumps(Model.model_json_schema())
-    build_regex_from_schema(json_schema, whitespace_pattern=None)
-
-
-def test_match_object():
-    test_regex = to_regex(
-        {
-            "type": "object",
-            "maxProperties": 0,
-        }
-    )
-    assert test_regex == r"\{[ ]?\}"
+    schema = json.dumps(FooBar.model_json_schema())
+
+    regex = build_regex_from_schema(schema)
+    expected = """{"foo" : 4 ,"bar":"baz    baz baz bar"}"""
+    assert re.fullmatch(regex, expected)
+
+    # any whitespace pattern can be used
+    regex = build_regex_from_schema(schema, r"[\n ]*")
+    expected = """{     "foo"   :   4, \n\n\n   "bar": "baz    baz baz bar"\n\n}"""
+    assert re.fullmatch(regex, expected)
+
+
+def test_invalid_json():
+    with pytest.raises(
+        TypeError,
+        match="Expected a valid JSON string.",
+    ):
+        build_regex_from_schema("{'name':")
+
+
+def test_types_presence():
```

Comment:
Not sure what this test is testing. If those are not defined, imports would error no? I think a better thing to do could be to test to whether these are defined as expected, by using the `re` module and trying to see if `EMAIL` regex can match an e-mail for example.

Reply:
Here we're only testing python interfaces, the presence of the types and not emptiness. I'll move import under the test to be clearer.
For the matching of many sorts we have 1000+ lines of tests in rust: https://github.com/dottxt-ai/outlines-core/blob/main/src/json_schema/mod.rs#L25-L1145